sympy/utilities/iterables.py

```diff
@@ -1381,6 +1381,12 @@ def partitions(n, m=None, k=None, size=False):
         raise ValueError("n must be >= 0")
     if m == 0:
         raise ValueError("m must be > 0")
+    if n == 0:
+        if size:
+            yield 1, {0: 1}
```

Comment:
@jksuom, I just learned that the convention is to return count of 1 if a partition cannot be done -- one empty set -- and SymPy follows that convention:

```
>>> from sympy.functions.combinatorial.numbers import nT
>>> nT(0, 0)
1
```

So perhaps it's best to return an empty dictionary as you said so `sum(1 for 1 in partitions(n,k))` will return the right value regardless of `n` and `k`.